docs/software_arch.md

```diff
@@ -0,0 +1,75 @@
+# Software Architecture
+
+This document provides an overview of the Maya submitter plug-in and adaptor that are in this repository.
+The intent is to help you have a basic understanding of what the applications are doing to give you context
+to understand what you are looking at when diving through the code. This is not a comprehensive deep dive of
+the implementation.
+
+## Maya Submitter Plug-in
+
+The Maya plug-in is contructed in two parts:
+1. A very bare-bones plug-in [module](https://help.autodesk.com/view/MAYAUL/2024/ENU/?guid=Maya_SDK_Distributing_Maya_Plug_ins_DistributingUsingModules_html)
+   in the `/maya_submitter_plugin` directory. 
+2. The `deadline.maya_submitter` Python package located in `/src/deadline/maya_submitter` that provides all of the actual business logic for the plugin.
+
+### Plug-in Module
+
+The plug-in module is mainly two files:
+
+1. `DeadlineCloudForMaya.mod` -- The [module description file](https://help.autodesk.com/view/MAYAUL/2024/ENU/?guid=Maya_SDK_Distributing_Maya_Plug_ins_DistributingUsingModules_ModuleDescriptionFiles_html)
+   for the plug-in.
+2. `plug-ins/DeadlineCloudForMaya.py` -- The implementation of Maya's plug-in interface for the plugin. There is not much to this code.
+   It is just building the shelf in Maya for AWSDeadline, and creating the button(s) in that shelf for the users to click to launch the submitter.
+   Clicking the submitter button calls a function in the `deadline.maya_submitter` package to create the UI, and do all of the submitter functionality.
+
+### `deadline.maya_submitter`
+
+This Python module implements all of the functionality of the Maya plug-in itself. The entrypoints that Maya integrates with
+are all defined in `deadline.maya_submitter.mel_commands`. These simply create a UI window to display to the user.
+
+Fundamentally, what this submitter is doing is creating a [Job Bundle](https://docs.aws.amazon.com/deadline-cloud/latest/developerguide/build-job-bundle.html)
+and using the GUI creation code in the [`deadline` Python package](https://pypi.org/project/deadline/) to generate the UI
+that is displayed to the user. The important parts to know about in a job bundle are:
+
+1. The job template file. The submitter code dynamically generates the template based on properties of the specific scene file
+   that is loaded. For example, it may contain a Step for each layer of the scene to render.
+   Note: All job template files are currently derived from a standard static job template located at
+   `src/deadline/maya_submitter/default_maya_job_template.yaml`.
+2. Asset references. These are the files that the job, when submitted, will require to be able to run. The submitter contains code
+   that introspects the scene that is loaded to automatically discover these, and the submitter UI allows the end-user to modify
```

Comment:
I don't quite follow the second sentence. Can we simplify it?

Reply:
Hah. Yeah, for sure. I wrote it and I can't even make sense of the sentence :-)